Remote paging with batch editing (#1966)
IgniteUI/igniteui-docfx: en/components/grids_templates/remote_data_operations.md

### Remote Paging with Batch editing

From the examples above should be clear how to set up the @@igxName with remote data. Now, let's enable batch editing for the grid following the [Batch Editing topic/guide](batch_editing.html).
Before continuing with the sample it is good to clarify the current use case. When we perform the pagination on the server side the grid contains the data only for the current page and if we add new rows the newly added rows will be concatenated with the current data that grid contains. If we try to go to a page and the server returns no data for that page the grid data source will be consisted only from the newly added rows, which grid will paginate according to defined pagination settings(page, perPage).

```typescript
public ngOnInit() {
  ...
    this._dataLengthSubscriber = this.remoteService.getDataLength().subscribe((data) => {
        this.totalCount = data;
        this._recordOnServer = data;
        this._totalPagesOnServer = Math.floor(this.totalCount / this.perPage);
        this.grid1.isLoading = false;
    });
    }
```

So in order to handle this use case properly we will implement a mix of remote and local pagination logic.

> First it is necessary to know the total number of the records that are on the server. Based on that information and how much records we would like the display per page we can calculate which is the last page with remote data and when to perform local pagination.

Reviewer: ```suggestion
Given that, we calculate the total number of data pages on the server (see `this._totalPagesOnServer `) and based on its value, we implement custom logic.
```

```typescript

public paginate(page: number) {
    this.grid1.endEdit(true);
    if (page > this._totalPagesOnServer) {
        if (this.page !== this._totalPagesOnServer) {
            const skipEl = this._totalPagesOnServer * this.perPage;
            this.remoteService.getData(skipEl, this.perPage);
        }
        this.grid1.page = page - this._totalPagesOnServer;
        this.page = page;
        return;
    } else {
        this.grid1.page = 0;
    }
    this.page = page;
    const skip = this.page * this.perPage;
    this.remoteService.getData(skip, this.perPage);
}

```

> As you can see in the **paginate** method based on the calculated value `this._totalPagesOnServer` we perform either remote pagination or local pagination by changing the @@igComponent `page` property.

Reviewer: ```suggestion
As you can see in the **paginate** method, custom pagination logic is performed, based on the `_totalPagesOnServer` value. 
```
